language/stdlib/modules/LibraVMConfig.move

```diff
@@ -3,15 +3,30 @@ address 0x1 {
 module LibraVMConfig {
     use 0x1::LibraConfig;
     use 0x1::Roles;
+    use 0x1::Vector;
+
+    const SCRIPT_HASH_LENGTH: u64 = 32;
+    const ESCRIPT_HASH_LENGTH_NOT_MATCH: u64 = 0;
 
     // The struct to hold all config data needed to operate the LibraVM.
     // * publishing_option: Defines Scripts/Modules that are allowed to execute in the current configruation.
     // * gas_schedule: Cost of running the VM.
     struct LibraVMConfig {
-        publishing_option: vector<u8>,
+        publishing_option: PublishingOption,
         gas_schedule: GasSchedule,
     }
 
+    struct PublishingOption {
+        // Only script hashes in the following list can be executed by the network.
+        script_allow_list: vector<vector<u8>>,
+        // VM can execute arbitrary script provided by the user if this flag is set to true.
+        is_open_script: bool,
+        // Transactions with sender in this list won't abide to the allow list rules above.
+        sender_allow_list: vector<address>,
+        // Anyone can publish new module if this flag is set to true.
+        is_open_module: bool,
```

Comment:
Nit: I find this name confusing without "publishing" in it; sounds like "openness" might be a quality of existing modules. Perhaps `module_publishing_allowed` or similar?

Reply:
+1 on this name changing 🥇